ember/app/services/fetch.js

```diff
@@ -0,0 +1,53 @@
+import Service, { inject as service } from "@ember/service";
+import { isEmpty } from "@ember/utils";
+import fetch from "fetch";
+
+const CONTENT_TYPE = "application/vnd.api+json";
+
+const cleanObject = (obj) => {
+  return Object.entries(obj).reduce((clean, [key, value]) => {
+    return {
+      ...clean,
+      ...(isEmpty(value) ? {} : { [key]: value }),
+    };
+  }, {});
+};
```

Comment:
Two minor things:
1. You can drop both returns since you're in an arrowfunction:
```suggestion
const cleanObject = (obj) =>
  Object.entries(obj).reduce((clean, [key, value]) => ({
    ...clean,
    ...(isEmpty(value) ? {} : { [key]: value }),
  }), {});
```
2. You could use `Object.fromEntries` here to reduce complexity and improve readability:
```suggestion
const cleanObject = (obj) => Object.fromEntries(
  Object.entries(obj).filter(([key, value]) => !isEmpty(value))
);
``````

Reply:
Applied 2nd suggestion